src/Type/Php/RegexArrayShapeMatcher.php

```diff
@@ -372,12 +372,13 @@ private function parseGroups(string $regex): ?array
 	{
 		if (self::$parser === null) {
 			/** @throws void */
-			self::$parser = Llk::load(new Read('hoa://Library/Regex/Grammar.pp'));
+			self::$parser = Llk::load(new Read(__DIR__ . '/../../../resources/RegexGrammar.pp'));
 		}
 
 		try {
 			$ast = self::$parser->parse($regex);
 		} catch (Exception) {
+			// TODO log parse errors in vebose mode?
```

Comment:
How can we achieve this from a matcher is it possible to output diagnostics/errors at least for easier debugging?

Reply:
Thanks that helps (tho it needs `$grammar  = new Hoa\File\Read('resources/RegexGrammar.pp');` now, but I meant more a way to just output `$e->getMessage()` so we can show the parse error if a regex fails to parse. That might belong more into a rule scanning regexes fed into preg_match for correctness though.